clients/cli/src/billing.rs

```diff
@@ -0,0 +1,173 @@
+            }
+            LbError::UiError(ExpiredCard) => CliError::billing("Card expired."),
+            LbError::UiError(ClientUpdateRequired) => CliError::update_required(),
+            LbError::UiError(CurrentUsageIsMoreThanNewTier) => {
+                CliError::billing("Your current usage exceeds the requested tier.")
+            }
+            LbError::UiError(ExistingRequestPending) => CliError::billing(
+                "Another billing request is being processed, please wait and try again later.",
+            ),
+            LbError::Unexpected(msg) => CliError::unexpected(msg),
+        })?;
+
+    println!("Account upgrade successful!");
+    Ok(())
+}
+
+fn cancel_subscription(core: &Core) -> Result<(), CliError> {
+    println!("Cancelling subscription...");
+    use lockbook_core::CancelSubscriptionError::*;
+    core.cancel_subscription().map_err(|err| match err {
+        LbError::UiError(NotPremium) => CliError::billing("You have no subscriptions to cancel!"),
+        LbError::UiError(AlreadyCanceled) => {
+            CliError::billing("This subscription has already been cancelled.")
+        }
+        LbError::UiError(UsageIsOverFreeTierDataCap) => CliError::billing(
+            "Your usage exceeds the trial tier, please delete excess files first.",
+        ),
+        LbError::UiError(ExistingRequestPending) => CliError::billing(
+            "Another billing request is being processed, please wait and try again later.",
+        ),
+        LbError::UiError(CouldNotReachServer) => CliError::network_issue(),
+        LbError::UiError(ClientUpdateRequired) => CliError::update_required(),
+        LbError::Unexpected(msg) => CliError::unexpected(msg),
+    })?;
+
+    println!("Subscription cancelled successfully!");
+    Ok(())
+}
+
+fn reuse_old(card: &str) -> Result<bool, CliError> {
+    Ok(Confirm::new()
+        .with_prompt(format!("Do you want use *{}?", card))
+        .interact()?)
+}
+
+fn solicit_card_info() -> Result<PaymentMethod, CliError> {
+    let number: String = Input::new()
+        .with_prompt("Enter your card number")
+        .interact_text()?;
+
+    let exp_year: i32 = Input::new()
+        .with_prompt("Expiration Year: ")
+        .interact_text()?;
+
+    let exp_month: i32 = Input::new()
+        .with_prompt("Expiration Month: ")
+        .interact_text()?;
+
+    let cvc: String = Input::new().with_prompt("CVC: ").interact_text()?;
+
+    Ok(PaymentMethod::NewCard { number, exp_year, exp_month, cvc })
+}
+
+fn get_subscription_info(core: &Core) -> Result<Option<SubscriptionInfo>, CliError> {
+    core.get_subscription_info().map_err(|err| match err {
+        LbError::UiError(GetSubscriptionInfoError::CouldNotReachServer) => {
+            CliError::network_issue()
+        }
+        LbError::UiError(GetSubscriptionInfoError::ClientUpdateRequired) => {
+            CliError::update_required()
+        }
+        LbError::Unexpected(msg) => CliError::unexpected(msg),
+    })
+}
```

Comment:
This is essentially just error mapping, so maybe instead impl `From<GetSubscriptionInfoError>` for CliError?

Reply:
bedbf2f4